### PR TITLE
chore(ci): auto-label based on PR title

### PR DESCRIPTION
**Pull request by jamilbk**
- Auto-labels PRs based on PR title so that they're automatically categorized when generating the Changelog.
- Enforce ConventionalCommit-style PR titles so they can be easily categorized in the changelog

### CODE REVIEW

**Comment by AndrewDryga** [comment]
Would it be better to use labels instead and make a CI step that makes sure that they are set? We can also auto-assigns component/* labels based on % of lines changed per toot folder

**Comment by jamilbk** [comment]
> Would it be better to use labels instead and make a CI step that makes sure that they are set?

Yeah thats another option, but the labels aren't shown in the changelog:

Screenshot 2024-03-02 at 3 21 45 PM

If a PR is opened without a label, and the check fails, would creating a label cause the check to run again? Maybe there's a "label changed" event trigger

**Comment by jamilbk** [comment]
The purpose behind this is really just to make my life easier editing the changelog and compiling release notes for socials 😅

File: .github/workflows/_static-analysis.yml [hunk 1]
```diff
@@ -3,6 +3,29 @@ on:
   workflow_call:
 
 jobs:
+  pr-title-check:
+    runs-on: ubuntu-22.04
```
Comment (thomaseizinger) [inline .github/workflows/_static-analysis.yml:8]: I have used https://github.com/marketplace/actions/semantic-pull-request before in other projects and it worked well.

**Review by thomaseizinger — COMMENTED** [review]
One suggestion on how to enforce this.

Also, I'd personally prefer to lower-case the title ("auto-label 
.." in your case) as described in conventional commits: https://www.conventionalcommits.org/en/v1.0.0/

**Comment by thomaseizinger** [comment]
If you ever wanted to group those changelog updates, then I think we should reserve "feat" and "fix" for _product_ related changes. Like in the case of this PR, I'd probably categorise it as "chore(ci)".

**Comment by jamilbk** [comment]
> If you ever wanted to group those changelog updates, then I think we should reserve "feat" and "fix" for _product_ related changes. Like in the case of this PR, I'd probably categorise it as "chore(ci)".

Good call -- I realized that right after opening this.

Feat and fix can be reserved for user facing features and bugfixes. This will allow us to point to the release when communicating relevant changes to users (or at least it's a start). People do read those.

File: .github/workflows/_static-analysis.yml (hunk 1 above)
Comment (jamilbk) [reply .github/workflows/_static-analysis.yml:8]: 👍

**Comment by ReactorScram** [comment]
"feat and fix can be reserved for..."

Sounds good, and that can go in CONTRIBUTING.md maybe in the Conventional Commits section

**Comment by jamilbk** [comment]
Adding @firezone/engineering to review just to give everyone a headsup. I may rename your PR titles before merge just to make the changelog a little more consumable for users.

**Review by ReactorScram — APPROVED** [review]
LGTM

**Pull request merged**